src/uu/pgrep/src/process_matcher.rs

```diff
@@ -411,8 +414,48 @@ fn test_parse_pidfile_content_valid() {
     assert_eq!(parse_pidfile_content("\n123\n"), None);
 }
 
-pub fn read_pidfile(filename: &str) -> UResult<i64> {
-    let content = fs::read_to_string(filename)
+#[cfg(unix)]
+fn check_pidfile_locked(file: &std::fs::File) -> bool {
```

Comment:
A detail: I would rename the function to `is_pidfile_locked` (or `is_locked`). Otherwise the condition on line 451, `if check_locked && !check_pidfile_locked(&file)`, reads a bit awkward: `check_locked` is about whether a lock check should be performed whereas `check_pidfile_locked` is about doing the lock check (and not about whether a different lock check should be performed).

Reply:
Indeed, I had it originally return UResult<()> so `check_` made more sense then but not for the predicate version. Changed to `is_`